File: solutions/swift/proverb/1/Sources/Proverb/Proverb.swift
class Proverb {
  private(set) var result = ""

  init(_ words: [String]) {
    guard !words.isEmpty else { return }
    for i in 1..<words.count {
      result += "For want of a \(words[i-1]) the \(words[i]) was lost.\n"
    }
    result += "And all for the want of a \(words[0])."
  }

  func recite() -> String { result }
}

File: solutions/ruby/proverb/1/proverb.rb
class Proverb
  def initialize(*args, **kwargs)
    @words = args
    @qualifier = kwargs.fetch(:qualifier, "")
  end

  def to_s
    verses = @words.each_cons(2).map { |w1, w2|
      "For want of a #{w1} the #{w2} was lost."
    }
    conclusion = [@qualifier, @words.first].join(' ').strip
    verses.join("\n") + "\nAnd all for the want of a #{conclusion}."
  end
end
